Feature/cannon commons

coreywoodfield opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/byu-oit-appdev/android-byu-suite-v2/63)
<!-- Reviewable:end -->

romrell4 commented:
Reviewed 1 of 25 files at r1.
Review status: 1 of 25 files reviewed at latest revision, 1 unresolved discussion.

---

*[app/src/main/AndroidManifest.xml, line 275 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSc_zE6cq86g_nk2g2v:-KSc_zE6cq86g_nk2g2w:bt7fy3y) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/AndroidManifest.xml#L275)):*
> ```XML
> 					android:name="is_feature"
> 					android:value="true" />
> 			<!--TODO: add this junk-->
> ```

Finish TODO

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: 0 of 25 files reviewed at latest revision, 3 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsCategoryActivity.java, line 36 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSca7SQgbFG3DpY6phj:-KSca7SQgbFG3DpY6phk:b-r8otvu) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsCategoryActivity.java#L36)):*
> ```Java
> 
> 		String title = getIntent().getStringExtra(CATEGORY);
> 		getSupportActionBar().setTitle(title.equals("Wall of Lettuce") ? "Salad Bar" : title);
> ```

Comment here

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java, line 40 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScaHA7v42GcgZ8_IKX:-KScaHA7v42GcgZ8_IKY:b9t0iyh) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java#L40)):*
> ```Java
> public class CannonCommonsMainActivity extends ByuActivity {
> 	private static final String HAS_VIEWED_DINING_LEGAL_INFO = "HAS_VIEWED_DINING_LEGAL_INFO";
> 	public static final String SERVE_DATE_FORMAT = "%d%02d%02d";
> ```

public? Is it used anywhere outside this activity?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

cwoodfie commented:
Review status: 0 of 25 files reviewed at latest revision, 1 unresolved discussion.

---

*[app/src/main/AndroidManifest.xml, line 275 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSc_zE6cq86g_nk2g2v:-KScaIuwsyfgtdvveunp:b-896fix) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/AndroidManifest.xml#L275)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Finish TODO
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: 0 of 25 files reviewed at latest revision, 4 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java, line 32 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScaSv1sz0RGXRoxckg:-KScaSv1sz0RGXRoxckh:b-kacq1b) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java#L32)):*
> ```Java
> import java.util.concurrent.*;
> 
> import static edu.byu.suite.features.cannonCommons.controller.CannonCommonsMapActivity.MEAL_ID;
> ```

Don't use the static import. It makes it difficult to see where those variables are coming from. Just import the classes that contain the tag

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: 0 of 25 files reviewed at latest revision, 5 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java, line 50 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScacroOFltCkvkChcE:-KScacroOFltCkvkChcF:btqektm) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java#L50)):*
> ```Java
> 	private static final int RECYCLER_VIEW = 1;
> 	private static final int TEXT_VIEW = 2;
> 	public static final SimpleDateFormat displayDateFormat = new SimpleDateFormat("EEE MMM d yyyy", Locale.US);
> ```

public? Used anywhere else

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: 0 of 25 files reviewed at latest revision, 6 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java, line 62 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScao5fdsE0hy2FIAjr:-KScao5fdsE0hy2FIAjs:bcvrlex) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java#L62)):*
> ```Java
> 		Calendar now = Calendar.getInstance();
> 		datePicker.init(now.get(Calendar.YEAR), now.get(Calendar.MONTH), now.get(Calendar.DAY_OF_MONTH), (datePicker1, year1, month1, day1) -> onDateChanged(year1, month1, day1));
> //		onDateChanged(now.get(Calendar.YEAR), now.get(Calendar.MONTH), now.get(Calendar.DAY_OF_MONTH));
> ```

Is there a reason this one is still in here? Let's put a comment before this init (and maybe in a comment or in person tomorrow explain to me what it's doing? Is it a listener?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: 0 of 25 files reviewed at latest revision, 7 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java, line 72 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScb9k5j7X4uxIzEcmx:-KScb9k5j7X4uxIzEcmy:b-1tgxli) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java#L72)):*
> ```Java
> 			new AlertDialog.Builder(this).setTitle("Legal Disclaimer")
> 										 .setMessage(R.string.dining_disclaimer)
> 										 .setNeutralButton("I understand", null)
> ```

Let's do positive here (not that it really matters in the long run)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: 0 of 25 files reviewed at latest revision, 8 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java, line 69 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScbHsltWGiAFDorQBP:-KScbHsltWGiAFDorQBQ:b-a88eok) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java#L69)):*
> ```Java
> 
> 		SharedPreferences preferences = getPreferences(MODE_PRIVATE);
> 		if (preferences.getBoolean(HAS_VIEWED_DINING_LEGAL_INFO, false)) {
> ```

Wait... If they HAVE viewed dining legal info, show the alert dialog?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: 0 of 25 files reviewed at latest revision, 9 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java, line 93 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScbTp-A1e0b02gKwj7:-KScbTp-A1e0b02gKwj8:b-emyn05) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java#L93)):*
> ```Java
> 	private void getMealId(int year, int month, int day) {
> 		viewAnimator.setDisplayedChild(PROGRESS_DIALOG);
> 		if (scheduledFuture != null) {
> ```

What is this doing?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 1 of 25 files at r1.
Review status: 1 of 25 files reviewed at latest revision, 9 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: 1 of 25 files reviewed at latest revision, 10 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 24 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScc6dfJxpO-FkeKmKd:-KScc6dfJxpO-FkeKmKe:b84trkj) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L24)):*
> ```Java
> import java.util.*;
> 
> import static edu.byu.suite.features.cannonCommons.controller.CannonCommonsCategoryActivity.CATEGORY;
> ```

Remove static imports

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

cwoodfie commented:
Review status: 1 of 25 files reviewed at latest revision, 10 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsCategoryActivity.java, line 36 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSca7SQgbFG3DpY6phj:-KScaWXbiSK5jLKgQtEZ:b-896fix) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsCategoryActivity.java#L36)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Comment here
</details>
Done.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java, line 32 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScaSv1sz0RGXRoxckg:-KScbCJHO2j54IEEr6ei:b-896fix) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java#L32)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Don't use the static import. It makes it difficult to see where those variables are coming from. Just import the classes that contain the tag
</details>
Done.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java, line 40 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScaHA7v42GcgZ8_IKX:-KScafkjHRgWWHdBOiYs:b-896fix) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java#L40)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> public? Is it used anywhere outside this activity?
</details>
Done.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java, line 50 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScacroOFltCkvkChcE:-KScbUzVfx5HLTW_oQew:b-896fix) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java#L50)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> public? Used anywhere else
</details>
Done.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java, line 62 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScao5fdsE0hy2FIAjr:-KScbe9puNQozD5hYXK9:b-365sww) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java#L62)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Is there a reason this one is still in here? Let's put a comment before this init (and maybe in a comment or in person tomorrow explain to me what it's doing? Is it a listener?
</details>
The only way to set an onDateChangedListener for a DatePicker is to do it with init. there's no setOnDateChangedListener(...)

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java, line 69 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScbHsltWGiAFDorQBP:-KScbx3iaaXU7e4tg7T-:b-oz8o77) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java#L69)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Wait... If they HAVE viewed dining legal info, show the alert dialog?
</details>
Good catch

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java, line 72 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScb9k5j7X4uxIzEcmx:-KScc-gKRzElCk6MwV5U:b-896fix) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java#L72)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Let's do positive here (not that it really matters in the long run)
</details>
Done.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java, line 93 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScbTp-A1e0b02gKwj7:-KScc3on4dtgeFjDgWzW:b-3v9gkt) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMainActivity.java#L93)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> What is this doing?
</details>
If a request has already been scheduled, cancel it and schedule the updated one. The onDateChanged gets called every time the date changes at all, so we don't want to get menu information for every date they pass through, just for the one they stop on

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 24 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScc6dfJxpO-FkeKmKd:-KSccM7rXVjjw5bBFSc2:b-896fix) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L24)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Remove static imports
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: 1 of 25 files reviewed at latest revision, 11 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 55 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSccVYgnrNzSnCiBIAM:-KSccVYgnrNzSnCiBIAN:b-v3nm8u) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L55)):*
> ```Java
> 
> 		Intent intent = getIntent();
> 		getSupportActionBar().setSubtitle(intent.getStringExtra(SUBTITLE));
> ```

Why is subtitle in ByuConstants?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: 0 of 25 files reviewed at latest revision, 12 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 64 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSccdYmJmH3Wg15f5az:-KSccdYmJmH3Wg15f5b-:b-8663u3) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L64)):*
> ```Java
> 			public void onSuccess(Call<MenuWrapper> call, Response<MenuWrapper> response) {
> 				menus = response.body().menus;
> 				for (MenuWrapper.Menu.Recipe recipe : menus.get(0).recipes) {
> ```

We only go through the first menu?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: 0 of 25 files reviewed at latest revision, 13 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 101 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScd5SpIzY5kisGTxYa:-KScd5SqA9mGh6ZcTmNt:b15c9tc) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L101)):*
> ```Java
> 		String category;
> 		switch (color) {
> 			case -1:
> ```

Oh wow! This is pretty ugly! I'm assuming this is straight from v1. What exactly is it doing, and is there a better way to do it?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

cwoodfie commented:
Review status: 0 of 25 files reviewed at latest revision, 13 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 55 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSccVYgnrNzSnCiBIAM:-KSccgHbJudPDpfIEmP6:b-y7mfrd) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L55)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Why is subtitle in ByuConstants?
</details>
It's used in all four activities, and i figure we might use it somewhere else, for convenience. i can move it though if you want

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 64 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSccdYmJmH3Wg15f5az:-KScdYCpoR47zTJ0vXWn:b-i18usj) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L64)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> We only go through the first menu?
</details>
There's only one menu. the service is silly.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 101 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScd5SpIzY5kisGTxYa:-KScdaeQyZFpQDhO5f-6:b2h6ui4) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L101)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Oh wow! This is pretty ugly! I'm assuming this is straight from v1. What exactly is it doing, and is there a better way to do it?
</details>
I think it takes the integer value of the color and switches it. lol. once you understand that it's pretty clear what it's doing. it just checks the color of the image at that point and does something based off that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

cwoodfie commented:
Review status: 0 of 25 files reviewed at latest revision, 13 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 101 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KScd5SpIzY5kisGTxYa:-KSceInzpBd6I9pnevcm:brx29v7) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/b1e651f897f69280ae48abf0a4b2731d4ed8d6c2/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L101)):*
<details><summary><i>Previously, cwoodfie (Corey Woodfield) wrote…</i></summary>
> I think it takes the integer value of the color and switches it. lol. once you understand that it's pretty clear what it's doing. it just checks the color of the image at that point and does something based off that.
</details>
I understand how getColorForPosition works, but not how getColorRoundTwo works (that's a fallback in case the first one doesn't work - i think the first one usually works though)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 19 of 25 files at r1, 7 of 7 files at r2.
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 125 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSftxsSI6meZEqxwOiP:-KSftxsSI6meZEqxwOiQ:b-3mhcve) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L125)):*
> ```Java
> 
> 	private int getColorForPosition(float x, float y) {
> 		ImageView background = (ImageView) findViewById(R.id.cannon_map_background);
> ```

Let's put a comment here to explain what this function is doing

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 129 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSfu4z2KZ3ZsG3LZsix:-KSfu4z2KZ3ZsG3LZsiy:b-x7113p) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L129)):*
> ```Java
> 		Bitmap drawingCache = background.getDrawingCache();
> 		if (drawingCache == null) {
> 			return getColorRoundTwo(x, y);
> ```

Put a log on this line, try to hit this page 50 times with as many phones as you can, and if you never get the log statement, let's remove it (my hunch is that it might be to support previous old versions that don't use a drawing cache or something)

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 159 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSfuS1y9Z6DNPkd8aZk:-KSfuS1y9Z6DNPkd8aZl:b-5jru9q) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L159)):*
> ```Java
> 	}
> 
> 	private static class Area {
> ```

Oh man there's got to be a better way to do this... Let's chat about this one. :-/

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsRecipeActivity.java, line 44 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSfun-qwkSYEHrlgMfz:-KSfun-qwkSYEHrlgMg-:b-qbfr4) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsRecipeActivity.java#L44)):*
> ```Java
> 		data.add(ALLERGY_INFORMATION);
> 		for (Map.Entry<String, Boolean> entry : recipe.allergens.entrySet()) {
> 			if (entry.getValue()) {
> ```

So what is this doing? Comments would be nice all throughout this function. Like LOTS of them.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/model/MenuWrapper.java, line 85 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgBQsXgp0sLH97KKoT:-KSgBQsXgp0sLH97KKoU:b-9cq1pa) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/model/MenuWrapper.java#L85)):*
> ```Java
> 	}
> 
> 	public static final String[] NUTRIENT_NAMES = {
> ```

Wasn't this passed in from the service? We don't want these hard coded

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/service/CannonCommonsClient.java, line 38 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgBcGiPIjt94rTY2nP:-KSgBcGjze54mBIf272G:bwm17t7) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/service/CannonCommonsClient.java#L38)):*
> ```Java
> 	public Converter.Factory getConverter() {
> 		BooleanFromIntAdapter booleanFromIntAdapter = new BooleanFromIntAdapter();
> 		return GsonConverterFactory.create(new GsonBuilder().setFieldNamingStrategy(f -> f.getName().toLowerCase())
> ```

What is this converter doing?

---

*[app/src/main/java/edu/byu/suite/features/personDirectory/service/AddressTypeAdapter.java, line 49 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgCFS03v83Z_gDUSYn:-KSgCFS03v83Z_gDUSYo:b-65lf70) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/personDirectory/service/AddressTypeAdapter.java#L49)):*
> ```Java
> 				break;
> 			case NULL:
> 				in.nextNull();
> ```

What were these changes about?

---

*[app/src/main/res/layout/cannon_commons_activity_category.xml, line 16 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgCg9qiOLXs6EDYaMP:-KSgCg9qiOLXs6EDYaMQ:btrw41o) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/res/layout/cannon_commons_activity_category.xml#L16)):*
> ```XML
> 			android:id="@+id/cannon_recipe_list"
> 			android:layout_width="0dp"
> 			android:layout_height="wrap_content"
> ```

Whenever you get time, show me this constraint layout stuff :)

---

*[app/src/main/res/layout/cannon_commons_activity_main.xml, line 19 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgD-Hi2idNrK4xBMgl:-KSgD-Hi2idNrK4xBMgm:b-i5yegm) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/res/layout/cannon_commons_activity_main.xml#L19)):*
> ```XML
> 			android:id="@+id/cannon_date_picker"
> 			android:layout_width="wrap_content"
> 			android:layout_height="0dp"
> ```

Why are we setting this to 0? Does it have to do with the constraint stuff?

---

*[support/src/main/java/edu/byu/support/ByuRecyclerView.java, line 15 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgHIBqjJWXQWq8sRrx:-KSgHIBrj6xxqJjdthbT:b-7nndvn) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/support/src/main/java/edu/byu/support/ByuRecyclerView.java#L15)):*
> ```Java
>  */
> public class ByuRecyclerView extends RecyclerView {
> 	// The first and third constructors aren't used (by us), but they're required when creating a custom view.
> ```

Gotcha. So does something break when we delete them? Seth couldn't seem to break the app...

---

*[support/src/main/java/edu/byu/support/utils/TextUtil.java, line 6 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgHeZXpBP9fbuCUCdB:-KSgHeZXpBP9fbuCUCdC:b8a7m3b) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/support/src/main/java/edu/byu/support/utils/TextUtil.java#L6)):*
> ```Java
>  * Created by cwoodfie on 9/13/16.
>  */
> public class TextUtil {
> ```

Let's rename this to TextUtils (as well as any other ones that are singular)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

cwoodfie commented:
Review status: 23 of 26 files reviewed at latest revision, 11 unresolved discussions, some commit checks failed.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 125 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSftxsSI6meZEqxwOiP:-KSmBeWjOoJQBzgGW2QH:b-896fix) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L125)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Let's put a comment here to explain what this function is doing
</details>
Done.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 129 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSfu4z2KZ3ZsG3LZsix:-KSmByq3pGjwhOlBrMfT:b-896fix) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L129)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Put a log on this line, try to hit this page 50 times with as many phones as you can, and if you never get the log statement, let's remove it (my hunch is that it might be to support previous old versions that don't use a drawing cache or something)
</details>
Done.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 159 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSfuS1y9Z6DNPkd8aZk:-KSmBzJDByFQI4U-82Vc:b-896fix) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L159)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Oh man there's got to be a better way to do this... Let's chat about this one. :-/
</details>
Done.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/model/MenuWrapper.java, line 85 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgBQsXgp0sLH97KKoT:-KSmAfyKhLAwMcrLjk8H:b-uzol10) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/model/MenuWrapper.java#L85)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Wasn't this passed in from the service? We don't want these hard coded
</details>
Yes, that array is no longer used, i just forgot to take it out.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/service/CannonCommonsClient.java, line 38 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgBcGiPIjt94rTY2nP:-KSmA6oJlpOfH4b5EPoc:b-2mmo7x) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/service/CannonCommonsClient.java#L38)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> What is this converter doing?
</details>
hehehe which part? the type adapter reads ints as booleans (0 = false, 1 = true). the field naming strategy changes all the java field names to lowercase, cus the json just uses lowercase (no underscores or anything). the excludeFieldsWithModifier, i'll explain to you next time we're both in at the same time. i don't think you'll like what it does very much. just remember, if it ain't broke, don't fix it.

---

*[app/src/main/java/edu/byu/suite/features/personDirectory/service/AddressTypeAdapter.java, line 49 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgCFS03v83Z_gDUSYn:-KSmAoOt7s1W7T_fLH_J:b-hayjf2) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/personDirectory/service/AddressTypeAdapter.java#L49)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> What were these changes about?
</details>
I looked back at it and realized i had implemented things slightly wrong, so i figured i'd just throw the fixes in here

---

*[app/src/main/res/layout/cannon_commons_activity_category.xml, line 16 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgCg9qiOLXs6EDYaMP:-KSmAxL7WvkJCndQLNfB:b-fq4lwm) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/res/layout/cannon_commons_activity_category.xml#L16)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Whenever you get time, show me this constraint layout stuff :)
</details>
will do :+1: 

---

*[app/src/main/res/layout/cannon_commons_activity_main.xml, line 19 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgD-Hi2idNrK4xBMgl:-KSmB-BMYOy9jEdqNxWq:b-wh7oi2) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/res/layout/cannon_commons_activity_main.xml#L19)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Why are we setting this to 0? Does it have to do with the constraint stuff?
</details>
yes. with constraint layouts, instead of saying match_parent, you say 0dp, which tells it to give it as much space as it can (something like that)

---

*[support/src/main/java/edu/byu/support/ByuRecyclerView.java, line 15 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgHIBqjJWXQWq8sRrx:-KSmB8M0qET_aX7tJybw:b-dvp9sl) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/support/src/main/java/edu/byu/support/ByuRecyclerView.java#L15)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Gotcha. So does something break when we delete them? Seth couldn't seem to break the app...
</details>
The only thing that breaks is that Android Studio no longer recognizes them as custom views, so when it tries to render a layout file, an error pops up saying it couldn't render the ByuRecyclerView. not a huge deal, but if you're trying to use the layout editor, it kind of is a huge deal

---

*[support/src/main/java/edu/byu/support/utils/TextUtil.java, line 6 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgHeZXpBP9fbuCUCdB:-KSmBNV-auTAbeL-EjoK:b2b92c1) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/support/src/main/java/edu/byu/support/utils/TextUtil.java#L6)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Let's rename this to TextUtils (as well as any other ones that are singular)
</details>
There's already an android class called TextUtils. i named it TextUtil so that you can use both in the same file without having to include the fully qualified domain name of one or the other. We could maybe call it StringUtils

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

cwoodfie commented:
Review status: 23 of 26 files reviewed at latest revision, 11 unresolved discussions, some commit checks failed.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsRecipeActivity.java, line 44 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSfun-qwkSYEHrlgMfz:-KSmC8JK_GQhzdoUV1hI:b-896fix) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsRecipeActivity.java#L44)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> So what is this doing? Comments would be nice all throughout this function. Like LOTS of them.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 3 of 7 files at r3.
Review status: 21 of 26 files reviewed at latest revision, 4 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 84 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSmJR8pk7DpHKwXtW9q:-KSmJR8pk7DpHKwXtW9r:b21g4mc) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L84)):*
> ```Java
> 		});
> 
> 		scaleX = (float) (findViewById(R.id.cannon_map).getMeasuredWidth() / 640.0);
> ```

Why did we remove these?

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsRecipeActivity.java, line 114 at r3](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSmKcUCgWqsDpOODGOA:-KSmKcUDjezFM2L5dWdv:bu9gqop) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/0a8d0555c1bcb0e668c927e27e6c4f5b20845626/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsRecipeActivity.java#L114)):*
> ```Java
> 					return new IngredientViewHolder(ViewUtils.inflate(parent, R.layout.default_list_item));
> 			}
> 			return null;
> ```

Will this give us any issues? I know we'll "never" get it, but if there were a null ByuViewHolder, what would happen?

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/service/CannonCommonsClient.java, line 38 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgBcGiPIjt94rTY2nP:-KSmKw7-GmwdA2Qg5Yj9:b-r7y7y3) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/service/CannonCommonsClient.java#L38)):*
<details><summary><i>Previously, cwoodfie (Corey Woodfield) wrote…</i></summary>
> hehehe which part? the type adapter reads ints as booleans (0 = false, 1 = true). the field naming strategy changes all the java field names to lowercase, cus the json just uses lowercase (no underscores or anything). the excludeFieldsWithModifier, i'll explain to you next time we're both in at the same time. i don't think you'll like what it does very much. just remember, if it ain't broke, don't fix it.
</details>
Lol. Well if it ain't built, build it right ;) I don't think I have too much of an issue with any of that, but I'd still love you to explain it to me sometime

---

*[support/src/main/java/edu/byu/support/utils/TextUtil.java, line 6 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgHeZXpBP9fbuCUCdB:-KSmLLhhiUsvFTbZePXl:bdbs20z) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/support/src/main/java/edu/byu/support/utils/TextUtil.java#L6)):*
<details><summary><i>Previously, cwoodfie (Corey Woodfield) wrote…</i></summary>
> There's already an android class called TextUtils. i named it TextUtil so that you can use both in the same file without having to include the fully qualified domain name of one or the other. We could maybe call it StringUtils
</details>
Ah. Yes. StringUtils sounds great :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 4 of 7 files at r3.
Review status: 25 of 26 files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

cwoodfie commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 84 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSmJR8pk7DpHKwXtW9q:-KSmQk_iAnlbvvOms0ud:b89c7bz) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L84)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Why did we remove these?
</details>
Because they weren't used anywhere

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsRecipeActivity.java, line 114 at r3](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSmKcUCgWqsDpOODGOA:-KSmPJYeYS0vhMzGMuBb:bwi1806) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/0a8d0555c1bcb0e668c927e27e6c4f5b20845626/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsRecipeActivity.java#L114)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Will this give us any issues? I know we'll "never" get it, but if there were a null ByuViewHolder, what would happen?
</details>
I'm not sure. It is 100% guaranteed that we'll never get it though, because viewType comes from getItemViewType, which will either return one of those three valid ints, or throw a runtime exception

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/service/CannonCommonsClient.java, line 38 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgBcGiPIjt94rTY2nP:-KSmPszRS7gczpDyumsx:boz3aog) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/service/CannonCommonsClient.java#L38)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Lol. Well if it ain't built, build it right ;) I don't think I have too much of an issue with any of that, but I'd still love you to explain it to me sometime
</details>
So when i removed the public static final array of nutrients, i made it a public static field in MenuWrapper. I made it static so that i wouldn't have to deal with passing it through a parcel. what `excludeFieldsWithModifiers(Modifier.TRANSIENT)` does is it tells gson to try to initialize static variables as well. so every time we make a request, it's redefined to be whatever nutrient array was passed back in the response (which, so far, has been the same every time), but since it's static, i don't need to pass it through the parcel, i can just access the most recent one from wherever i happen to be at the time. :smile: 

---

*[support/src/main/java/edu/byu/support/utils/TextUtil.java, line 6 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgHeZXpBP9fbuCUCdB:-KSmPmkDHfhe5UfF7oHC:b-896fix) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/support/src/main/java/edu/byu/support/utils/TextUtil.java#L6)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Ah. Yes. StringUtils sounds great :)
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 2 of 3 files at r5.
Review status: 25 of 26 files reviewed at latest revision, 1 unresolved discussion.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/service/CannonCommonsClient.java, line 38 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgBcGiPIjt94rTY2nP:-KSmRkf8HJwx2lvHZVCy:b-uozlqo) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/service/CannonCommonsClient.java#L38)):*
<details><summary><i>Previously, cwoodfie (Corey Woodfield) wrote…</i></summary>
> So when i removed the public static final array of nutrients, i made it a public static field in MenuWrapper. I made it static so that i wouldn't have to deal with passing it through a parcel. what `excludeFieldsWithModifiers(Modifier.TRANSIENT)` does is it tells gson to try to initialize static variables as well. so every time we make a request, it's redefined to be whatever nutrient array was passed back in the response (which, so far, has been the same every time), but since it's static, i don't need to pass it through the parcel, i can just access the most recent one from wherever i happen to be at the time. :smile: 
</details>
Oh man... Well, do you think this is the best way to do it? If so, I'll merge it

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 1 of 3 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

cwoodfie commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java, line 42 at r5](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSmSTh8C1ymfJUfA37a:-KSmSTh8C1ymfJUfA37b:b-yhj0hx) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/4cb53407c011f8664499e91fd8d21973dbc01060/app/src/main/java/edu/byu/suite/features/cannonCommons/controller/CannonCommonsMapActivity.java#L42)):*
> ```Java
> 		setContentView(R.layout.cannon_commons_activity_map);
> 		ProgressDialog progressDialog = ByuProgressDialog.createAndShowProgressDialog(this);
> 		// If they can cancel it by touching outside, they'll be able to click on areas before the list of menu items is ready, which will take them to an empty activity.
> ```

Have you looked at this?

What was happening was that if someone cancelled the progress dialog and then clicked somewhere on the map, the recipes wouldn't be ready yet, so an empty category activity would be pulled up. This doesn't let them cancel the progress dialog by clicking outside, so that they can't click on anything until the request is done. They can still go back, but if they hit back, it doesn't just cancel the dialog, it completely closes the activity.

I considered a few ways of ensuring that they can't click on the map until the request is done, and this seemed like the best way to do it. I just want to make sure you're okay with this implementation before you merge it

---

*[app/src/main/java/edu/byu/suite/features/cannonCommons/service/CannonCommonsClient.java, line 38 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSgBcGiPIjt94rTY2nP:-KSmS-0HQZsYTsF3IWaj:brwn4yi) ([raw file](https://github.com/byu-oit-appdev/android-byu-suite-v2/blob/08776a62dcffda9aed51c5b3ea7db7ae213ea7df/app/src/main/java/edu/byu/suite/features/cannonCommons/service/CannonCommonsClient.java#L38)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Oh man... Well, do you think this is the best way to do it? If so, I'll merge it
</details>
I think it's the easiest way to do it. I don't think there's any risk to it, as the only time we access that field is in the CannonCommonsRecipeActivity. The MenuWrapper (containing the static nutrient array) is gotten from the service in the CannonCommonsMapActivity. there's no way to get to the recipe activity without going through the map activity. so i can't think of any instance in which it wouldn't work

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

cwoodfie commented:
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

---

Review status: 23 of 26 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-:-KSn6g-bXSqkNYrk6OQ7:b-uwkowd)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 3 of 3 files at r7.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSrInNbbDDJxeDqRMmn:-KSrInNbbDDJxeDqRMmo:bwacnks) (no related file):*
Fix issues found by Scott's test

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

coreywoodfield commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSrInNbbDDJxeDqRMmn:-KSwKeECXz7jlMhda1XA:b-vrqts) (no related file):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Fix issues found by Scott's test
</details>
I fixed those on wednesday

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

coreywoodfield commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSrInNbbDDJxeDqRMmn:-KSwRS0pCmRhtAKK8Oxb:b-hryou1) (no related file):*
<details><summary><i>Previously, coreywoodfield (Corey Woodfield) wrote…</i></summary>
> I fixed those on wednesday
</details>
Oh wait i just saw your slack message nevermind i'll do that

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 12 of 13 files at r8.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSrInNbbDDJxeDqRMmn:-KTAbXwQr6JtxFUXxFIc:b-xaohy) (no related file):*
<details><summary><i>Previously, coreywoodfield (Corey Woodfield) wrote…</i></summary>
> Oh wait i just saw your slack message nevermind i'll do that
</details>
Is it all done?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->

coreywoodfield commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63#-KSrInNbbDDJxeDqRMmn:-KTAivqWqdFpmEPGK4U6:by9kd18) (no related file):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary>
> Is it all done?
</details>
I fixed the following issues:

1. a. We can’t really control the implementation of the DatePicker, unless we want to write our own. I don’t; you can if you want. The DatePicker does behave much better in marshmallow and lollipop than it does in kitkat, but there’s not much we can do about that.
2.a,b. fixed
  c. updated to match v1 more closely
3.a. I figure it’s better to tell the user that there’s no info for that day than to tell them nothing.

semicolons: added :, :
order of allergens: doesn’t seem important enough that I care to do anything about it

*TL;DR*
i fixed everything

```if ((Collection) (everything i mentioned above)).containsAll((Collection) all) {
	return true;
}```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/android-byu-suite-v2/63)*
<!-- Sent from Reviewable.io -->
